webui/react/src/components/DeterminedAuth.tsx

```diff
@@ -101,21 +102,23 @@ const DeterminedAuth: React.FC<Props> = ({ canceler }: Props) => {
         ]}>
         <Input
           autoFocus
+          data-testid="username"
           placeholder="username"
           prefix={<Icon name="user" size="small" title="Username" />}
         />
       </Form.Item>
       <Form.Item name="password">
         <Input.Password
+          data-testid="password"
           placeholder="password"
           prefix={<Icon name="lock" size="small" title="Password" />}
         />
       </Form.Item>
       {isBadCredentials && (
-        <p className={[css.errorMessage, css.message].join(' ')}>Incorrect username or password.</p>
+        <p className={[css.errorMessage, css.message].join(' ')} data-testid="error">Incorrect username or password.</p>
```

Comment:
nitpick: maybe `bad-cred-error` or something in case there is ever more than one on the page?

Reply:
there actually are two alerts since there's almost always one in the bottom right. I'd like to have a list of errors. Maybe a `BaseComponents` plural class?

Reply:
A list of errors would be nice. Feel free to punt to out of scope and make a follow up ticke here though since it sounds like that might be more involved. I don't love plural classes in languages with generic types. We can just do a Array<BaseComponent> and get most of the functionality for a lot less code.

Reply:
actually, this isn't even the notification that shows in our test. im gonna delete this change